### PR TITLE
Improve Reader class

### LINKED ISSUE #97: Improve Reader class
_From @Irineu333 in https://github.com/Irineu333/SpeakTouch/issues/91#issuecomment-1796526503_

### PR DESCRIPTION
**Pull request by PatrykMis**
### Improvements

* SpeakTouch now reads statusses of various controlls according to their roles.

### Fixed issues

* CheckedText was not announced as checked if checked

### Other

- Merge #108

### CODE REVIEW

File: app/src/main/java/com/neo/speaktouch/model/Type.kt [hunk 1]
```diff
@@ -63,6 +65,9 @@ enum class Type {
             // View->TextView->EditText
             if (className `is` EditText::class.java) return EDITFIELD
 
+            // View->TextView->CheckedTextView
+            if (className `is` CheckedTextView::class.java) return CHECKEDTEXT
```
Comment (Irineu333) [inline app/src/main/java/com/neo/speaktouch/model/Type.kt]: Good, I saw that it inherits from `Checkable`. I think we need to support it as well, to cover custom views.

Comment (PatrykMis) [reply app/src/main/java/com/neo/speaktouch/model/Type.kt]: This is the special case. checkedTextView is usually list item and should be announced only if checked (selected, marked, ...). If we generalize by `isCheckable` property, state would be spoken for each item, not only one checked which is not intentional behavior for just one checked item in the list view.

Comment (Irineu333) [reply app/src/main/java/com/neo/speaktouch/model/Type.kt]: Good, I wasn't aware of this particular behavior.

**Review by Irineu333 — APPROVED** [review]
Simply gorgeous

**Pull request merged**